pkg/registry/core/service/ipallocator/allocator.go

```diff
@@ -85,13 +88,16 @@ func NewAllocatorCIDRRange(cidr *net.IPNet, allocatorFactory allocator.Allocator
 	max := utilnet.RangeSize(cidr)
 	base := utilnet.BigForIP(cidr.IP)
 	rangeSpec := cidr.String()
+	var family api.IPFamily
 
 	if utilnet.IsIPv6CIDR(cidr) {
+		family = api.IPv6Protocol
 		// Limit the max size, since the allocator keeps a bitmap of that size.
 		if max > 65536 {
 			max = 65536
 		}
 	} else {
+		family = api.IPv4Protocol
 		// Don't use the IPv4 network's broadcast address.
```

Comment:
low pri. We should have a note here that we depend mask size (in config ref: https://github.com/kubernetes/kubernetes/blob/master/cmd/kube-apiserver/app/options/validation.go#L89) validation. if validation is removed then we need to max it to `max(unsiged short)`

Reply:
that is done by utilnet.RangeSize in L88 https://github.com/kubernetes/utils/blob/4a2b36d8d73f0bfa1092e740605b66db34fca1cd/net/net.go#L192-L204